streams/src/main/java/org/apache/kafka/streams/processor/internals/RecordCollectorImpl.java

```diff
@@ -81,7 +81,7 @@ public class RecordCollectorImpl implements RecordCollector {
     private final Sensor droppedRecordsSensor;
     private final Map<String, Sensor> producedSensorByTopic = new HashMap<>();
 
-    private final AtomicReference<KafkaException> sendException;
+    private final AtomicReference<KafkaException> sendException = new AtomicReference<>(null);
```

Comment:
There is no reason to get this from `StreamsProducer` so pulling it into this class.